pihole

```diff
@@ -98,8 +99,38 @@ versionFunc() {
   exit 0
 }
 
+# Get PID of main pihole-FTL process
+getFTLPID() {
+  local pid
+
+  if [ -s "${FTL_PID}" ]; then
+    # -s: FILE exists and has a size greater than zero
+    pid="$(<"$FTL_PID")"
+    # Verify that the value read from the file is numeric
+    if [[ "$pid" =~ [^[:digit:]] ]]; then
+      # If no, unset the variable so we use pgrep in the next step
+      pid=
+    fi
+  fi
+
+  # If no PID file is available, try to get a PID from pgrep
+  if [ -z "$pid" ]; then
+    # Use --oldest to get the main process of FTL (all TCP workers are created
+    # after the main process has been started)
+    pid="$(pgrep --oldest pihole-FTL)"
+  fi
+
+  # If we still have no PID, FTL is not running, substitute negative PID to
+  # signal this to the caller
+  if [ -z "$pid" ]; then
+    pid="-1"
+  fi
+
+  echo "$pid"
+}
+
```

Comment:
This uses some bashisms and really should be POSIX, but you know....

```suggestion
# Get PID of main pihole-FTL process
getFTLPID() {
  local ftl_pid
 
  if [ -s "${FTL_PID_FILE}" ]; then
    # -s: FILE exists and has a size greater than zero
    ftl_pid="$(<"$FTL_PID_FILE")"
    # Verify that the value read from the file is numeric
    if [[ "$ftl_pid" =~ [^[:digit:]] ]]; then
      # If pid value is not numeric then try to get a PID from pgrep
      # Use --oldest to get the main process of FTL (all TCP workers are created
      # after the main process has been started) OR use sentinel value to show pgrep failure
      ftl_pid="$(pgrep --oldest pihole-FTL)" || ftl_pid="-1"
    fi
  fi
  
  # echo the pid value found or -1 if variable is still unset
   echo "${ftl_pid:=-1}"
}

```

Reply:
The proposed change would change behavior as we'd never try to get the PID through `pgrep` if the PID file is absent (or empty).

Reply:
I pushed yet another commit addressing the issue that a system with a missing PID file is problematic.